front/app/containers/Admin/reporting/components/ReportBuilderPage/CreateReportModal.tsx

```diff
@@ -5,34 +5,80 @@ import { createReport } from 'services/reports';
 
 // components
 import Modal from 'components/UI/Modal';
-import { Box, Title, Text, Input } from '@citizenlab/cl2-component-library';
+import {
+  Box,
+  Title,
+  Text,
+  Input,
+  Label,
+  Radio,
+} from '@citizenlab/cl2-component-library';
 import Button from 'components/UI/Button';
+import ProjectFilter from 'containers/Admin/dashboard/components/filters/ProjectFilter';
 
 // styling
 import { colors } from 'utils/styleUtils';
 
+// utils
+import clHistory from 'utils/cl-router/history';
+
 // i18n
 import messages from './messages';
-import { FormattedMessage } from 'utils/cl-intl';
+import { FormattedMessage, MessageDescriptor } from 'utils/cl-intl';
+import { IOption } from 'typings';
 
 interface Props {
   open: boolean;
   onClose: () => void;
 }
 
+type Template = 'blank' | 'project';
+
+interface RadioLabelProps {
+  message: MessageDescriptor;
+}
+
+const RadioLabel = ({ message }: RadioLabelProps) => (
+  <Text mt="0px" mb="0px" variant="bodyS" color="primary">
+    <FormattedMessage {...message} />
+  </Text>
+);
+
 const CreateReportModal = ({ open, onClose }: Props) => {
   const [reportTitle, setReportTitle] = useState('');
+  const [template, setTemplate] = useState<Template>('blank');
+  const [selectedProject, setSelectedProject] = useState<string | undefined>();
   const [loading, setLoading] = useState(false);
   const reportTitleTooShort = reportTitle.length <= 2;
 
+  const blockSubmit =
+    reportTitleTooShort ||
+    (template === 'project' ? selectedProject === undefined : false);
+
+  const toggleTemplate = () => {
+    setTemplate((template) => (template === 'blank' ? 'project' : 'blank'));
+  };
+
+  const handleProjectFilter = (option: IOption) => {
+    setSelectedProject(option.value === '' ? undefined : option.value);
+  };
+
   const onCreateReport = async () => {
-    if (reportTitleTooShort) return;
+    if (blockSubmit) return;
     setLoading(true);
 
     try {
-      await createReport(reportTitle);
+      const report = await createReport(reportTitle);
       setLoading(false);
-      onClose();
+
+      const route = '/admin/reporting/report-builder';
+      const path = `${route}/${report.data.id}/editor`;
+      const params =
+        template === 'project' && selectedProject
+          ? `?projectId=${selectedProject}`
+          : '';
+
+      clHistory.push(path + params);
```

Comment:
Opens the report with correct query string containing project id if template was chosen

Reply:
Project ID is only used by the template though?